### PR TITLE
React capstone

### PR DESCRIPTION
**Pull request by tinatsina**
## Summary
- created an Air Quality SPA
- used the Air Quality China API
- can display each city in its own route

## Tech used
- react-icons
- CreateAsyncThunk
- Redux
- useEffect Hooks

## Deployment
- deployed using the Netlify app

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  const state = useSelector((state) => state.cityReducer);
+
+  useEffect(() => {
+    if (state.length === 0) dispatch(fetchCityRating(cityList));
+  }, [dispatch, state.length]);
+
+  return (
+    <>
+      <nav>
+        <p>Highest Polluters</p>
```
Comment (tufoinnkuo10) [inline src/components/Home/Home.js:23]: - [ ] please kindly try for your Home page to have a list of categories that could be filtered using given parameters, e.g. most polluted/least polluted

File: src/redux/reducers/cityReducer.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { FETCH_CITY_DATA } from '../actions/aqiActions';
+
+const initialState = [];
+
```
Comment (tufoinnkuo10) [inline src/redux/reducers/cityReducer.js:4]: - [ ] please kindly add some integration tests using React Testing Library and some unit tests for several components of the app and some pure functions in Redux code. Great work 😄

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @tinatsina 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: README.md [hunk 3]
```diff
@@ -1,70 +1,62 @@
-# Getting Started with Create React App
+![](https://img.shields.io/badge/Microverse-blueviolet)
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# China Air Quality Application
```
[the remark this answers is not in the document]
Comment (tinatsina) [reply README.md:3]: ## Resolved 👍
- used your solution and added the video
- its a LOOM video
- contains all the specifications
- thanks for assisting 🧡

File: src/redux/reducers/cityReducer.js (hunk 2 above)
Comment (tinatsina) [reply src/redux/reducers/cityReducer.js:4]: ## Resolved : Tests added
- added multiple tests including
    -snapshot
    -pure component test
    -API mocks

File: README.md [hunk 4]
```diff
@@ -1,70 +1,62 @@
...
+## Show your support
 
-### Analyzing the Bundle Size
+Give a ⭐️ if you like this project!
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size](https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size)
+## Acknowledgments
 
-### Making a Progressive Web App
+- Hat tip to Microverse
```
[the remark this answers is not in the document]
Comment (tinatsina) [reply README.md]: ## Resolved 👍 
- added the proper credits and licenses
- included the tribute to `Nelson Sakwa`
- thanks for noticing

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @tinatsina , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

### Highlights:
 ✔️ Project is setup correctly
 ✔️ All linters are passing
✔️ Correct gitflow
 ✔️ Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

image

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@shoirata] in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**